src/main/java/server/router/routes/Route.java

```diff
@@ -41,29 +49,36 @@ private void setStatus() {
     }
   }
 
-  private String setStartLine() {
+  private void setStartLine() {
     setStatus();
 
     StringBuilder startLine = new StringBuilder();
     startLine.append(Protocol.DEFAULT);
     startLine.append(status);
 
-    return startLine.toString();
+    this.startLine = startLine.toString();
   }
 
-  private ArrayList<String> setHeaders() {
+  private void setHeaders() {
```

Comment:
Methods `setHeaders`, `setStatus` and possibly others are starting to grow each time you add a new route. It looks like OCP violation. Have you taken a look at the real routing implementation in [Spring](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-ann-requestmapping) or [Phoenix](https://hexdocs.pm/phoenix/routing.html#examining-routes)? The main idea is that for each route, they provide some Handler or Controller who knows how to handle the request on a specific path. You implemented this using inheritance, but what if you try to use an interface for this? And each Handler/Controller can implement your `processRequest()` method, for example? Using this approach, you can prevent modifying existing code every time you add support for a route, implement a new handler, and build a response without condition statements.

Reply:
Thanks for sending this info! I did think about how Rails uses controllers but haven't considered implementing it using an interface... makes a lot of sense to me though! I'll work on this now

Reply:
Great! Let me know if you need help with this.

Reply:
Updated with a new `RouteHandler` interface. I'm not sure if I did this correctly -- now my abstract `Route` class implements the interface but my individual routes are still subclassing `Route`. It did allow me to remove all of my conditional statements but I wasn't sure if there was a more efficient way to route without using a combination of an interface and inheritance?